File: src/main/java/pl/mjedynak/idea/plugins/builder/gui/displayer/AbstractPopupDisplayer.java
package pl.mjedynak.idea.plugins.builder.gui.displayer;

import com.intellij.openapi.editor.Editor;
import javax.swing.JList;
import org.jetbrains.annotations.NotNull;
import pl.mjedynak.idea.plugins.builder.factory.PopupChooserBuilderFactory;

public abstract class AbstractPopupDisplayer {

    public void displayPopupChooser(@NotNull Editor editor, @NotNull JList<?> list, @NotNull Runnable runnable) {
        PopupChooserBuilderFactory.getPopupChooserBuilder(list)
                .setTitle(getTitle())
                .setItemChosenCallback(runnable)
                .setMovable(true)
                .createPopup()
                .showInBestPositionFor(editor);
    }

    @NotNull
    protected abstract String getTitle();
}

File: src/main/kotlin/io/github/kingg22/kobra/builder/gui/displayer/AbstractPopupDisplayer.kt
package io.github.kingg22.kobra.builder.gui.displayer

import com.intellij.openapi.editor.Editor
import io.github.kingg22.kobra.builder.factory.PopupChooserBuilderFactory
import javax.swing.JList

public abstract class AbstractPopupDisplayer {
    public fun displayPopupChooser(editor: Editor, list: JList<*>, runnable: Runnable) {
        PopupChooserBuilderFactory.getPopupChooserBuilder(list)
            .setTitle(this.title)
            .setItemChosenCallback(runnable)
            .setMovable(true)
            .createPopup()
            .showInBestPositionFor(editor)
    }

    protected abstract val title: String
}
